dashboard/config/routes.rb

```diff
@@ -324,6 +324,14 @@
     end
   end
 
+  resources :programming_environments, param: 'name' do
+    resources :programming_expressions, param: 'programming_expression_key' do
+      member do
+        get :show, to: 'programming_expressions#show_by_keys'
+      end
+    end
+  end
```

Comment:
beautiful! this looks way better than what i was suggesting. shorter names would be nice, but not at the expense of clarity.

Reply:
Agreed. I just spent way longer than I should've debating this URL and settled on this for now as it just makes sense for these models. I figure we can add a more user-friendly path if we need to in the future, if just using the /docs/ path that we will reuse anyway doesn't fit all our needs